src/main/java/de/tum/in/www1/artemis/service/hestia/structural/StructuralTestCaseService.java

```diff
@@ -125,14 +127,16 @@ public StructuralTestCaseService(GitService gitService, ProgrammingExerciseTestC
      * @return The new structural solution entries
      */
     private List<ProgrammingExerciseSolutionEntry> generateStructuralSolutionEntries(Set<ProgrammingExerciseTestCase> testCases, Repository solutionRepository,
-            StructuralClassElements[] classElements, Map<String, JavaClass> solutionClasses) {
-        return Arrays.stream(classElements).flatMap(classElement -> {
+            StructuralClassElements[] classElements, Map<String, JavaClass> solutionClasses) throws URISyntaxException {
+        List<ProgrammingExerciseSolutionEntry> programmingExerciseSolutionEntries = new ArrayList<>();
+
+        for (var classElement : classElements) {
             var packageName = classElement.getStructuralClass().getPackageName();
             var name = classElement.getStructuralClass().getName();
             var solutionClass = solutionClasses.get(packageName + "." + name);
             String filePath;
             if (solutionClass != null) {
-                filePath = solutionClass.getSource().getURL().getPath().replace(solutionRepository.getLocalPath().toAbsolutePath() + "/", "");
+                filePath = solutionRepository.getLocalPath().toAbsolutePath().toUri().relativize(solutionClass.getSource().getURL().toURI()).toString();
```

Comment:
This really should not be done in this PR. This change means that the Structural Solution Entry generation also has to be tested to check that this does not have any unintended side effects.
If you want to I can implement this in my Structural Solution Entry follow-up PR that I am currently working on.